### PR TITLE
chore: refactor cache client auth flow

### PR DESCRIPTION
**Pull request by Harasz**
### Contributor checklist

- [x] Breaking changes - check for any existing interfaces changes that are not backward compatible, removed method etc.
- [x] Documentation - document your code, add comments for method, remember to check if auto generated docs were updated.
- [x] Tests - add new or updated existed test for changes you made.
- [x] Migration guide - add migration guide for every breaking change.
- [x] Configuration correctness - check that any configuration changes are correct ex. default URLs, chain ids, smart contract verification on [Volta explorer](https://volta-explorer.energyweb.org/) or [EWC explorer](https://explorer.energyweb.org/).

### CODE REVIEW

**Review by jrhender — COMMENTED** [review]
Seems like a reasonable start 👍 

One question I have is whether or not it is important to preserve the order of the execution of the failed requests?
Previously, we were doing `this.failedRequests = this.failedRequests.filter((callback) => callback());`. I think this likely would have run the requests in the order of the `failedRequests` array. In the new `promiseRetry` approach, we have several requests awaiting the same promise, but it's not clear to me which will be selected when the promise resolved. So if there were multiple database updates, they could be out of order, for instance.
Does this analysis seem correct?

**Comment by Harasz** [comment]
It does not matter in my opinion. The execution of the failed request is manged by the code itself. We are operating in the same context of the promise. So if we have something like this 
```
await update(1);
await update(2);
```
and update 1 will fail, it will try to run it 3 times and eventually it will fail, but update 2 will not execute, bc first update throw an error. So from the code perspective its the same promise that needs to be fulfilled.

@jrhender

File: src/modules/cache-client/cache-client.service.ts [hunk 1]
```diff
@@ -565,15 +566,95 @@ export class CacheClient implements ICacheClient {
...
+    } finally {
+      this.isAuthenticating = false;
+    }
+
+    return true;
+  }
+
+  private async makeRetryRequest<T>(request: () => Promise<T>): Promise<T> {
+    return promiseRetry(
+      async (retry) => {
```
[the remark this answers is not in the document]
Comment (Harasz) [reply src/modules/cache-client/cache-client.service.ts:633]: I don't think so, thanks for catch it

**Review by JGiter — APPROVED** [review]
Interesting approach. May be it is possible to wrap axios API with promiseRetry in order to not repeat in every method

**Comment by Harasz** [comment]
@artursudnik request interceptor is only accepting axios config, so i can't override promise there. I can use response interceptor and catch errors, but it's not an ideal approach in my opinion.

**Pull request merged**